openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py

```diff
@@ -0,0 +1,177 @@
+from openedx.adg.common.course_meta.tests.factories import CourseMetaFactory
+from openedx.adg.lms.applications.management.commands import update_is_prerequisite_courses_passed as command_module
+from openedx.adg.lms.applications.models import ApplicationHub
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+
+
+@pytest.fixture
+def user_for_testing(request):
+    return UserFactory.create()
+
+
+@pytest.fixture
+def course_for_testing(request):
+    prerequisite_course = CourseOverviewFactory.create(
+        end_date=datetime(2022, 1, 1),
+        start_date=datetime(2015, 1, 1),
+        display_name='pre requisite course'
+    )
+
+    CourseMetaFactory(course=prerequisite_course, is_prereq=True)
+
+    return prerequisite_course
+
+
+@pytest.fixture
+def enroll_user(request, user_for_testing, course_for_testing):
+    CourseEnrollmentFactory.create(
+        user=user_for_testing,
+        course_id=course_for_testing.id,
+        is_active=request.param['is_active']
+    )
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': False})], indirect=True)
+def test_minimal_users_with_inactive_user(enroll_user):
+    """
+    Test to check that an inactive user is not counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+def test_minimal_users_none_registered_in_prerequisite(course_for_testing):
+    """
+    Test to check that a user not registered in prerequisite is not counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': True})], indirect=True)
+def test_minimal_users(enroll_user):
+    """
+    Test to check that an active user registered in prerequisite is counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() > 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': True})], indirect=True)
+def test_user_failed_in_course(user_for_testing, enroll_user):
+    """
+    Test to check if the helper function returns True for a user not passed in course
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
```

Comment:
This is not required in this test `pre_req_courses = CourseMeta.open_pre_req_courses.all()`